src/git/cache.go

```diff
@@ -0,0 +1,63 @@
+package git
+
+// StringCache caches a string value.
+// The zero value is an empty cache.
+type StringCache struct {
+	value       string
+	initialized bool
+}
+
+// Set allows collaborators to signal when the current branch has changed.
+func (sc *StringCache) Set(newValue string) {
+	sc.value = newValue
+	sc.initialized = true
+}
+
+// Value provides the current value.
+func (sc *StringCache) Value() string {
+	if !sc.initialized {
+		panic("using current branch before initialization")
```

Comment:
This error message seems wrong. Same for string slice.

Reply:
Thanks, good catch!